### ISSUE TITLE
Remove support for older PHP versions

### ISSUE THREAD

**Issue by finwe** [mpdf/mpdf#1637]
# Feature request

In mPDF version 10.0, older PHP versions support will be removed. Concrete versions are to be determined with regard to composer versions statistics which currenly are as such:

![obrazek](https://user-images.githubusercontent.com/195675/161930953-3db2843d-1b0e-4f96-b54b-0bb33d82cea5.png)

**Comment by jakejackson1** [comment]
Composer is one metric to look at, but it's always going to be skewed towards projects using more modern tooling, and therefore more likely to keep PHP updated.

Over in WordPress land ([which powers 43% of all websites, as listed by w3techs.com](https://w3techs.com/technologies/details/cm-wordpress)), the figures are skewed towards earlier versions of the 7.* branch: https://wordpress.org/about/stats/

![Screen Shot 2022-04-07 at 8 13 41 am](https://user-images.githubusercontent.com/2918419/162081158-56781142-b033-432a-b543-36675796e0ef.png)

**Comment by finwe** [comment]
Yes, but as Composer is our primary platform to distribute code, I believe it is safe to play it with their numbers. Users of older PHP will most likely not have the latest up-to-date mPDF.

**Comment by jakejackson1** [comment]
That's not quite accurate. I myself use the latest version of mPDF via composer and distribute it in a WordPress plugin to a not insubstantial ~50,000 active websites. About ~25,000 of which are running mPDF 8 and PHP7.3 or higher. Unfortunately I don't have a breakdown of specific PHP versions for my particular users; I only know the mPDF 8/PHP7 3 statistics because a bit over 50% is running the latest major version of the software (which includes mPDF8 and requires PHP7.3 or higher). 

My point is that we should take the composer statistic figures with a grain of salt and give a little extra weight/consideration (but not much) to older versions of PHP when looking at a new minimum PHP version. 

You also probably want to look at the PHP version statistics for the 8.0 release, rather than the lifetime stats. Even better would be to compare the usage for v8 from a year ago to now and see where the movement is, and how many downloads occurred during that time.

**Current stats for v8 from 11 months ago (couldn't quite get stats for 12 months ago)**
<img width="814" alt="Screen Shot 2022-04-07 at 4 08 08 pm" src="https://user-images.githubusercontent.com/2918419/162131544-3be15e48-6555-475d-92a1-69e91200bf75.png">

**Current stats for v8 from today**
<img width="916" alt="Screen Shot 2022-04-07 at 4 04 39 pm" src="https://user-images.githubusercontent.com/2918419/162131347-2fde7b95-509d-4c47-999a-552c802bf821.png">

From this period mPDF was downloaded from composer 177,400 times. 

| Version | Movement | Total Downloads |
| --- | ----------- | ------------ |
| 5.6        | -2.10% | 3193 |
| 7.0        | -0.40% | 2128 |
| 7.1         | -1.10% | 3725 |
| 7.2        | -5.50% | 11,885 |
| 7.3        | -12.80% | 19,336 |
| 7.4        | +4.30% | 93,844 |
| 8.0        | +8.00% |  26,077 |
| 8.1        | +9.60% |  17,207 |

**Comment by finwe** [comment]
You install composer libraries with a different PHP version then you run them on? Bold. How do you bypass the version mismatch check in the autoloading?

Well anyway, there is a whole another major version until this happens and version 9 should offer many of the enhancements 10 will while still supporting PHP 5.6.

As was noted in other threads, even 7.4 is now security updates only and I bet it will be unsupported when 10.0 comes out. And when doing a change as impactful as this, let's do it the most _right_ way.

**Comment by jakejackson1** [comment]
> You install composer libraries with a different PHP version then you run them on?

You lock composer [into the lowest version of PHP you plan on supporting](https://github.com/GravityPDF/gravity-pdf/blob/development/composer.json#L8-L10). It doesn't cause the autoloader any problems.

> Well anyway, there is a whole another major version until this happens and version 9 should offer many of the enhancements 10 will while still supporting PHP 5.6.
> As was noted in other threads, even 7.4 is now security updates only and I bet it will be unsupported when 10.0 comes out. And when doing a change as impactful as this, let's do it the most right way.

Sounds good. When it gets closer to mPDF 10 and a PHP version bump I'll redo the PHP stats from composer for the previous 12 months and update with the latest WP states. 👍

**Comment by finwe** [comment]
Reference Q3/2024

![obrazek](https://github.com/mpdf/mpdf/assets/195675/9d997037-2644-4476-9a60-f003203c5200)

**Comment by jakejackson1** [comment]
It's nice to see so many mPDF servers now running PHP 8.1+! @finwe do you have a specific percentage in mind that you'd like to see PHP7.4 fall to before dropping support?

**Comment by finwe** [comment]
@jakejackson1 No, I just might wake up one day and do it :)
